CONTRIBUTING.md

```diff
@@ -0,0 +1,32 @@
+# Contribution Guide For Developers
+
+## How To Contribute
+
+Thank you for your interest in contributing. As we aim to make it as easy as possible to contribute changes, his guide will introduce basic guidelines on code, documentation and tests that contributors must follow to ensure contributions are accepted quickly.
+
+## Pull Request
+
+The changes should be submitted through 'Pull Request'. Don't send your patches through other channels like email or slack. The submitted PR will be reviewed by the other developers and maintainers and it will be merged by 'Squash Merge'.
+
+We introduce several simple rules for your PR.
+
+* Follow the [7 rules of a great commit message](https://chris.beams.io/posts/git-commit/)
+* When fixing bugs, add tests
+* When providing performance fixes, describe your benchmarking approach so that we can reproduce it
+* Don't include user or tool specific config files or folder, such as .vimrc, .vscode .idea.
+* Any pull request submitted must build and pass tests before being merged.
```

Comment:
Trailing dot (`.`) on this line and the previous line, which are inconsistent with the previous bullet points